File: backend/models/InterviewSession.js
const mongoose = require('mongoose');

const interviewSessionSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    aptitudeResult: {
      correct: { type: Number, default: 0 },
      total: { type: Number, default: 0 },
      totalScore: { type: Number, default: 0 },
      categoryScores: { type: Map, of: Object, default: {} }
    },
    codingResult: {
      solved: { type: Number, default: 0 },
      total: { type: Number, default: 0 },
      avgScore: { type: Number, default: 0 },
      results: { type: Array, default: [] }
    },
    technicalResult: {
      overallScore: { type: Number, default: 0 },
      totalScore: { type: Number, default: 0 }
    },
    overallScore: {
      score: { type: Number, default: 0 },
      outOf: { type: Number, default: 90 },
      percent: { type: Number, default: 0 }
    },
    violations: { type: Number, default: 0 },
    disqualified: { type: Boolean, default: false },
    completedAt: { type: Date, default: Date.now }
  },
  { timestamps: true }
);

module.exports = mongoose.model('InterviewSession', interviewSessionSchema);

File: backend/models/interview_session.py
"""models/interview_session.py — InterviewSession table (replaces Mongoose InterviewSession.js)"""
from sqlalchemy import Column, Integer, Boolean, Float, DateTime, ForeignKey, JSON
from sqlalchemy.sql import func
from database import Base


class InterviewSession(Base):
    __tablename__ = "interview_sessions"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    aptitude_result = Column(JSON, default=dict)   # {correct, total, totalScore, categoryScores}
    coding_result = Column(JSON, default=dict)     # {solved, total, avgScore, results}
    technical_result = Column(JSON, default=dict)  # {overallScore, totalScore}
    overall_score = Column(JSON, default=dict)     # {score, outOf, percent}
    violations = Column(Integer, default=0)
    disqualified = Column(Boolean, default=False)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
